src/inference/Inference.jl

```diff
@@ -356,6 +356,9 @@ function AbstractMCMC.bundle_samples(
         info = NamedTuple()
     end
 
+    # Ensure that the chain is not of type Any.
+    parray = eltype(parray) == Any ? convert(Array{Union{Real, Missing}}, parray) : parray
```

Comment:
Actually, I don't think this is a proper solution that should be part of Turing since there's no way for a user to correct this in case `Array{Union{Real,Missing}}` is not a suitable type. It's not guaranteed to work, and hence I think something like `map(identity, parray)` (which is included above already?) or `concretize` is the best you can do in such a general function. As long as the type inference problems are not fixed where they arise, IMO users should cast to the correct types themselves afterwards by adding a line such as `concretize(chains)` or `convert`. I think we should rather add a `convert(MCMCChains{T}, ...)` method to MCMCChains that would just cast the underlying array to the correct new type than forcing `Array{Union{Real,Missing}}` in all cases where type inference fails.